pyteal/ast/block.py

```diff
@@ -0,0 +1,80 @@
+from typing import TYPE_CHECKING
+from enum import Enum
+
+from pyteal.types import TealType, require_type
+from pyteal.errors import verifyFieldVersion
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+from pyteal.ast.leafexpr import LeafExpr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class BlockField(Enum):
+    # fmt: off
+    #                 id  |    name     |      type     | min version
+    block_seed =      (0, "BlkSeed",      TealType.bytes,  7)  # noqa: E222
+    block_timestamp = (1, "BlkTimestamp", TealType.uint64, 7)  # noqa: E222
+
+    # fmt: on
+
+    def __init__(self, id: int, name: str, type: TealType, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.ret_type = type
+        self.min_version = min_version
+
+    def type_of(self) -> TealType:
+        return self.ret_type
+
+
+BlockField.__module__ = "pyteal"
+
+
+class Block(LeafExpr):
+    """An expression that accesses a block property."""
+
+    def __init__(self, field: BlockField, block: Expr) -> None:
+        super().__init__()
+        self.field = field
```

Comment:
do we need to check that field is indeed a BlockField?

Reply:
My assumption was that `Block` wouldn't be used directly but rather through the `Block.timestamp()` methods. I'm not opposed, I was just following the convention I'd seen in other files. I previously had an explicit check in `JsonRef` because passing the property was the expected behavior in that case.